src/kernel.cr

```diff
@@ -1,18 +1,39 @@
 {% if flag?(:win32) %}
+  # The standard input file descriptor.
   STDIN  = IO::FileDescriptor.new(0)
+  # The standard output file descriptor. Gets flushed when a newline is written.
   STDOUT = IO::FileDescriptor.new(1).tap { |f| f.flush_on_newline = true }
+  # The standard error file descriptor. Gets flushed when a newline is written.
   STDERR = IO::FileDescriptor.new(2).tap { |f| f.flush_on_newline = true }
 {% else %}
   require "c/unistd"
 
+  # The standard input file descriptor.
   STDIN  = IO::FileDescriptor.from_stdio(0)
+  # The standard output file descriptor. Gets flushed when a newline is written.
   STDOUT = IO::FileDescriptor.from_stdio(1).tap { |f| f.flush_on_newline = true }
+  # The standard error file descriptor. Gets flushed when a newline is written.
   STDERR = IO::FileDescriptor.from_stdio(2).tap { |f| f.flush_on_newline = true }
 {% end %}
 
+# The name of the current program.
```

Comment:
This is actually the name the program was called by under the shell. For example `/foo/bar arg` in the shell gets `/foo/var` as `PROGRAM_NAME`, whereas just `bar` in the path gets just `bar` as `PROGRAM_NAME`.